src/Raven.Server/Documents/ETL/EtlProcess.cs

```diff
@@ -338,6 +344,17 @@ public IEnumerable<TTransformed> Transform(IEnumerable<TExtracted> items, Docume
                         stats.RecordChangeVector(item.ChangeVector);
                         stats.RecordLastFilteredOutEtag(item.Etag, item.Type);
 
+                        item.Dispose();
+
+                        if (extractedItemsSize % (Database.Configuration.Etl.MaxNumberOfExtractedItems ?? 8192) == 0)
+                        {
+                            if (CanContinueBatch(stats, item, extractedItemsSize, context) == false)
```

Comment:
In here, are we accounting for the size of the transaction (encryption, compression buffer, etc)?

Reply:
Yes, this is accounted here:
`...LowLevelTransaction.AdditionalMemoryUsageSize`


https://github.com/ravendb/ravendb/blob/19a3cb12b806157bce90a7e89a6a952b4666b1ee/src/Raven.Server/Documents/ETL/EtlProcess.cs#L546